extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -369,6 +387,92 @@ export class AmpLightboxViewer extends AMP.BaseElement {
       this.previous_();
     }
   }
+
+  /**
+   * Display gallery view to show thumbnails of lightboxed elements
+   * @private
+   */
+  openGallery_() {
+    // Build gallery div for the first time
+    if (!this.gallery_) {
+      this.buildGallery_();
+    }
+    this.container_.setAttribute('gallery-view', '');
+  }
+
+  /**
+   * Close gallery view
+   * @private
+   */
+  closeGallery_() {
+    this.container_.removeAttribute('gallery-view');
+  }
+
+  /**
+   * Build lightbox gallery. This is called only once when user enter gallery
+   * view for the first time.
+   * @private
+   */
+  buildGallery_() {
+    // Build gallery
+    this.gallery_ = this.win.document.createElement('div');
+    this.gallery_.classList.add('-amp-lightbox-viewer-gallery');
+
+    // Initialize thumbnails
+    this.updateThumbnails_();
```

Comment:
gallery_ node isn't attached to anything when building thumbnails, does it make sense not to do vsync in `this.updateThumbnails_` and instead do `this.container_.appendChild(this.gallery_);` here in a vsync? @dvoytenko @zhouyx